Use dark theme in GridTileBar

cbracken opened the pull request:
Reduce IconTheme, Typography hardcoding by applying a dark theme to the
contents. This is still not ideal; the theme brightness should really be
driven by backgroundColor.

cbracken committed:
Use dark theme in GridTileBar

Reduce IconTheme, Typography hardcoding by applying a dark theme to the
contents. This is still not ideal; the theme brightness should really be
driven by backgroundColor.

Hixie commented on packages/flutter/lib/src/material/grid_tile_bar.dart:
```diff
@@ -118,9 +121,8 @@ class GridTileBar extends StatelessWidget {
-        context: context,
-        data: new IconThemeData(color: Colors.white),
+      child: new Theme(
+        data: darkTheme,
```
does this really work? I am surprised this is equivalent.

cbracken replied:
Thanks for spotting -- you're entirely correct; it's not. TIL that the `IconThemeData` associated with a `Theme` doesn't override any `IconTheme` above it. 

While I put together a separate change to refactor `Theme` & `IconTheme`, I've patched this to add back an IconTheme, but eliminate the typography hardcoding.

cbracken committed:
Add back IconTheme

Since IconTheme.of() finds the closest IconTheme, not the closest
IconTheme or Theme (which specifies IconThemeData), the previous commit
doesn't necessarily override any IconThemes futher up the tree.

Hixie commented:
![LGTM](http://junkyard.damowmow.com/563)
